mountpoint-s3/src/prefetch.rs

```diff
@@ -331,7 +324,7 @@ where
             to_read -= part_len;
         }
 
-        Ok(response)
+        Ok(response.into_bytes()?)
```

Comment:
not relevant to this PR changes, but for the sake of understanding: so we validate the checksum here, at which point is it computed? I see that we combine checksums of `part_bytes`, is it the `S3CrtClient` (our code) who computes checksum of the `part_bytes` in the first place? or it's provided by S3?

Reply:
We do it in`mountpoint-s3`, here when downloading from S3: https://github.com/awslabs/mountpoint-s3/blob/608b921cb7b81d2012cd196087aa6aeaeaa1b6b1/mountpoint-s3/src/prefetch/part_stream.rs#L222